distributed/tests/test_worker.py

```diff
@@ -1665,6 +1665,32 @@ async def test_pip_install_fails(c, s, a, b):
 #             assert args[1:] == ["-m", "pip", "--upgrade", "install", "requests"]
 
 
+@gen_cluster(client=True)
+async def test_upload_file_plugin(c, s, a, b):
+    def g():
+        import myfile
+
+        return myfile.x
+
+    with tmp_text("myfile.py", "x = 123") as fn:
+        await c.register_worker_plugin(UploadFile(fn))
+        x = await c.submit(g)
+
+        assert x == 123
+
+
+@gen_cluster(client=True)
+async def test_upload_file_plugin_exception(c, s, a, b):
+    def g():
+        import myfile
+
+        return myfile.x
+
+    with tmp_text("myfile.py", "!syntax_error") as fn:
+        with pytest.raises(SyntaxError):
+            await c.register_worker_plugin(UploadFile(fn))
```

Comment:
I'm curious if there is ever an occasion when we would want users to interact with `UploadFile` directly, rather than using `Client.upload_file`.  If not, then we might want to only use the `upload_file` method in tests and treat `UploadFile` as an implementation detail.  Historically I've found that this makes it easier to change internals in the future (as you've just done here).  Thoughts?

Reply:
The only difference I can think of is that the user could give the plugin a different name, thereby changing the behavior of re-uploading the same file multiple times. I've chosen `fname + uuid` for the plugin name to allow re-uploads to take effect.

I'm not sure why one might want to do that, however.

Reply:
Yeah, I'm inclined to stick with just the upload_file method.  This PR is also fine as is though if you want to be done.  I'm mostly splitting hairs here.

Reply:
I love deleting code